tools/odbc/driver.cpp

```diff
@@ -49,6 +49,7 @@ SQLRETURN SQLFreeHandle(SQLSMALLINT handle_type, SQLHANDLE handle) {
 	}
 	case SQL_HANDLE_STMT: {
 		auto *hdl = (duckdb::OdbcHandleStmt *)handle;
+		hdl->dbc->stmt_handle = nullptr;
```

Comment:
Should we check if the statement is actually the current statement, in case there are multiple active statements?

Reply:
What I understood so far is that such situation is not possible because it implies in having two open statement handles, like:

```
SQLAllocHandle(SQL_HANDLE_STMT, conn, &hstmt);
SQLAllocHandle(SQL_HANDLE_STMT, conn, &hstmt2);
```

Am I right? Or did you mean another situation?

Reply:
Yes, exactly. It is not possible to have multiple open streaming query results inside the C++ API, but that does not necessarily mean it is impossible to have two open ODBC statements. Is that correctly checked for? What happens if the above code is executed and the user tries to open two statements at the same time?